src/solc_0.8/polygon/child/land/PolygonLandBaseToken.sol

```diff
@@ -356,6 +356,35 @@ contract PolygonLandBaseToken is ERC721BaseToken {
         return ownerOfAll || _owners[quadId] == uint256(uint160(address(from)));
     }
 
+    function _ownerOf(uint256 id) internal view override returns (address) {
+        require(id & LAYER == 0, "Invalid token id");
+        uint256 x = id % GRID_SIZE;
+        uint256 y = id / GRID_SIZE;
+        uint256 owner1x1 = _owners[id];
+
+        if (owner1x1 != 0) {
+            return address(uint160(owner1x1)); // cast to zero
```

Comment:
del the comment and add a comment to the method to explain this conditions waterfall